src/test/java/com/datastax/oss/protocol/internal/request/AuthResponseTest.java

```diff
@@ -43,16 +41,33 @@ public AuthResponseTest() {
   @Test
   @UseDataProvider(location = TestDataProviders.class, value = "protocolV3OrAbove")
   public void should_encode_and_decode(int protocolVersion) {
```

Comment:
maybe we should rename test to `should_encode_and_decode_and_clear_token`?

Reply:
Clearing the token is a side-effect of encoding, sounds clear enough to me.